src/libmongoc/src/mongoc/mongoc-scram.c

```diff
@@ -1038,6 +1038,8 @@ _mongoc_sasl_prep_impl (const char *name, const char *in_utf8, bson_error_t *err
    }
 
    /* convert to unicode. */
+   BSON_ASSERT (num_chars < SSIZE_MAX);
+   BSON_ASSERT ((num_chars + 1) <= SIZE_MAX / sizeof (uint32_t));
```

Comment:
```suggestion
   BSON_ASSERT (bson_cmp_less_equal_su (num_chars, SIZE_MAX / sizeof (uint32_t) - 1u));
```
Avoid integer comparison signedness warnings.